mlir/lib/Conversion/ConvertToAIRPass.cpp

```diff
@@ -470,12 +470,16 @@ Value lookupOrDefaultRange(Value v, IRMapping &remap) {
   return remap.lookupOrDefault(v);
 }
 
-Operation *getLinalgOpFromExecuteOp(Operation *op) {
-  Operation *output = nullptr;
+Operation *getCoreComputeOpFromExecuteOp(Operation *op) {
+  // We assume all linalg ops (except for linalg.copy) and func.call ops do
+  // computations only and do not participate in data movement.
   if (auto exec = dyn_cast<air::ExecuteOp>(op)) {
-    exec.walk([&](linalg::LinalgOp linalg_op) { output = linalg_op; });
+    if (isa<linalg::LinalgOp>(exec.getChildOp()))
+      return exec.getChildOp();
+    else if (isa<func::CallOp>(exec.getChildOp()))
+      return exec.getChildOp();
```

Comment:
I think this is a great point. For now we do asynchronous analysis on every single operation, so that the `air::ExecuteOp` currently never contains more than one operation in its body. But I think the definition for execute op is not restricted to single operation. So perhaps we should revisit the `getChildOp` method to either enforce to single child only, or do a `getChildOps` instead.